src/mainboard/emulation/qemu-q35/src/main.rs

```diff
@@ -0,0 +1,65 @@
+#![feature(asm)]
+#![feature(lang_items, start)]
+#![no_std]
+#![no_main]
+#![feature(global_asm)]
+
+use core::fmt::Write;
+use core::panic::PanicInfo;
+use model::Driver;
+use payloads::payload;
+use print;
+use uart::ns16550::NS16550;
+use wrappers::{Memory, SectionReader, SliceReader};
+
+global_asm!(include_str!("../../../../arch/x86/x86_64/src/bootblock.S"));
+//global_asm!(include_str!("init.S"));
+
+#[no_mangle]
+pub extern "C" fn _start(fdt_address: usize) -> ! {
+    let uart0 = &mut NS16550::new(0x10000000, 115200);
```

Comment:
Hmmm, somehow this has to use port 3f8. I'm not sure if the register package supports port io.

Reply:
we'll take that one on later.